user-interface/src/main/java/life/qbic/datamanager/views/projects/project/samples/registration/batch/BatchRegistrationDialog2.java

```diff
@@ -0,0 +1,547 @@
+package life.qbic.datamanager.views.projects.project.samples.registration.batch;
+
+import static java.util.Objects.isNull;
+import static java.util.function.Function.identity;
+
+import com.vaadin.flow.component.AbstractField.ComponentValueChangeEvent;
+import com.vaadin.flow.component.ClickEvent;
+import com.vaadin.flow.component.ComponentEvent;
+import com.vaadin.flow.component.ComponentEventListener;
+import com.vaadin.flow.component.button.Button;
+import com.vaadin.flow.component.html.Div;
+import com.vaadin.flow.component.html.Span;
+import com.vaadin.flow.component.icon.VaadinIcon;
+import com.vaadin.flow.component.textfield.TextField;
+import com.vaadin.flow.data.renderer.ComponentRenderer;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+import java.util.StringJoiner;
+import java.util.stream.Collectors;
+import life.qbic.datamanager.views.general.DialogWindow;
+import life.qbic.datamanager.views.general.spreadsheet.Spreadsheet;
+import life.qbic.datamanager.views.general.spreadsheet.Spreadsheet.ValidationMode;
+import life.qbic.datamanager.views.projects.project.samples.registration.batch.BatchRegistrationDialog2.ConfirmEvent.Data;
+import life.qbic.projectmanagement.domain.model.experiment.BiologicalReplicate;
+import life.qbic.projectmanagement.domain.model.experiment.Condition;
+import life.qbic.projectmanagement.domain.model.experiment.ExperimentalGroup;
+import life.qbic.projectmanagement.domain.model.experiment.vocabulary.Analyte;
+import life.qbic.projectmanagement.domain.model.experiment.vocabulary.Species;
+import life.qbic.projectmanagement.domain.model.experiment.vocabulary.Specimen;
+import life.qbic.projectmanagement.domain.model.sample.AnalysisMethod;
+
+/**
+ * TODO!
+ * <b>short description</b>
+ *
+ * <p>detailed description</p>
+ *
+ * @since <version tag>
+ */
+public class BatchRegistrationDialog2 extends DialogWindow {
+
+  private final Spreadsheet<SampleInfo> spreadsheet;
+
+  private final TextField batchNameField;
+
+  private final List<ExperimentalGroup> experimentalGroups;
+  private final List<Species> species;
+  private final List<Specimen> specimen;
+  private final List<Analyte> analytes;
+
+
+  public BatchRegistrationDialog2(String experimentName,
```

Comment:
NIT: The constructor does a lot of work here, maybe the individual initialization could be split up into distinct methods(initSpreadsheet, initSpreadsheetControls etc.) so it's more readable 🤔

Reply:
nope makes it harder to read. This can all be in the same scope IMHO.